clients/portal/portal.go

```diff
@@ -0,0 +1,114 @@
+// Package portal is a client module to support server-side use of the Diabeloop
+// service called user-api.
+package portal
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/tidepool-org/go-common/clients/disc"
+)
+
+// Client is the interface to portal-api.
+type Client struct {
+	config     *ClientConfig
+	hostGetter disc.HostGetter
+	httpClient *http.Client
+}
```

Comment:
You should introduce an interface in order to mock your client easily in other libs using it.

And quickly watching what is done in other package in go-common the "Client" type is the interface and the "ServicenameClient" type is the struct (see what's done in shoreline.go). 

```suggestion
type Client interface {
	GetPatientConfig(token string) (*PatientConfig, error) 
}

type PortalClient struct {
	config     *ClientConfig
	hostGetter disc.HostGetter
	httpClient *http.Client
}
```